master/internal/command/command.go

```diff
@@ -146,357 +139,326 @@ func remakeCommandsByType(
+	DefaultCmdService.unregisterCommand(c.taskID)
+}
 
-	case *apiv1.KillTensorboardRequest:
-		err := task.DefaultService.Signal(c.allocationID, task.KillAllocation, "user requested kill")
-		if err != nil {
-			ctx.Log().WithError(err).Warn("failed to kill allocation")
-		}
-		ctx.Respond(&apiv1.KillTensorboardResponse{Tensorboard: c.toTensorboard(ctx)})
+func (cs *commandService) unregisterCommand(id model.TaskID) {
+	cs.mu.Lock()
+	defer cs.mu.Lock()
 
-	case *apiv1.SetTensorboardPriorityRequest:
-		err := c.setPriority(int(msg.Priority), true)
-		if err != nil {
-			ctx.Respond(err)
-			return nil
-		}
-		ctx.Respond(&apiv1.SetTensorboardPriorityResponse{Tensorboard: c.toTensorboard(ctx)})
-
-	case *apiv1.DeleteWorkspaceRequest:
-		if c.Metadata.WorkspaceID == model.AccessScopeID(msg.Id) {
-			err := task.DefaultService.Signal(
-				c.allocationID,
-				task.KillAllocation,
-				"user requested workspace delete",
-			)
-			if err != nil {
-				ctx.Log().WithError(err).Warn("failed to kill allocation while deleting workspace")
-			}
-		}
+	delete(cs.commands, id)
+}
 
-	case sproto.ContainerLog:
+// commandService NTSC methods: LaunchNTSC, GetNTSC, listNTSCType, DeleteWorkspaceNTSC
 
-	case terminateForGC:
-		ctx.Self().Stop()
+// launchNTSC launches a generic NTSC command & returns the command.
+// launchNTSC -> CreateGenericCommand -> startCmd() ...
+func (cs *commandService) launchNTSC(
+	ctx context.Context,
+	req *CreateGeneric,
+	jobType model.JobType,
+	taskType model.TaskType,
+) (*command, error) {
+	taskID := model.NewTaskID()
+	req.Spec.CommandID = string(taskID)
 
-	case sproto.SetGroupWeight:
-		err := c.SetWeight(msg.Weight)
-		if err != nil {
-			ctx.Log().WithError(err).Info("setting command job weight")
-		}
-		if ctx.ExpectingResponse() {
-			ctx.Respond(err)
-		}
+	cmd, err := cs.createGenericCommand(ctx, taskID, taskType, model.NewJobID(),
+		jobType, req.Spec, req.ContextDirectory)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to launch %s", taskType)
+	}
+	return cmd, nil
+}
 
-	case sproto.SetGroupPriority:
-		err := c.setPriority(msg.Priority, true)
-		if err != nil {
-			ctx.Log().WithError(err).Info("setting command job priority")
-		}
-		if ctx.ExpectingResponse() {
-			ctx.Respond(err)
-		}
+// getNTSC gets & checks type of a command given its ID.
+func (cs *commandService) getNTSC(cmdID model.TaskID, cmdType model.TaskType) (*command, error) {
+	c, ok := cs.commands[cmdID]
+	if !ok {
+		return nil, fmt.Errorf("get NTSC %s not found", cmdID)
+	}
 
-	case sproto.SetResourcePool:
-		ctx.Respond(fmt.Errorf("setting resource pool for job type %s is not supported", c.jobType))
+	if c.taskType == cmdType {
+		return c, nil
+	}
+	cs.syslog.Debugf("getNTSC: type mismatch: %s/%s", cmdType, c.taskType)
+	return nil, fmt.Errorf("get NTSC %s not found", cmdID)
+}
 
-	default:
-		return actor.ErrUnexpectedMessage(ctx)
+// listNTSCType returns a list of NTSCs of one type.
+func (cs *commandService) listNTSCType(
```

Comment:
```suggestion
func (cs *commandService) listByType(
```
personally, i dislike "NTSC".